R/get_pvalue.R

```diff
@@ -74,21 +74,22 @@ get_pvalue <- function(survfit_object,
 
   survdifflist <- list(
     `Log-Rank`    = rlang::expr(eval(as.call(
-      append(NewCall, list(rho = 0))
+      append(!!NewCall, list(rho = 0))
     ))),
     `Wilcoxon`    = rlang::expr(eval(as.call(
-      append(NewCall, list(rho = 1))
+      append(!!NewCall, list(rho = 1))
     ))),
     `Tarone-Ware` = rlang::expr(eval(as.call(
-      append(NewCall, list(rho = 1.5))
+      append(!!NewCall, list(rho = 1.5))
     ))),
     `Custom`      = rlang::expr(eval(as.call(
-      append(NewCall, list(rho = rho))
+      append(!!NewCall, list(rho = !!rho))
     )))
   )[ptype]
 
 
-  survdifflist_eval <- lapply(survdifflist, eval, env = environment())
+  survdifflist_eval <-
```

Comment:
We only wish to evaluate what is required. It is to be more efficient.

Reply:
I didn't try to improve efficiency here, just modified existing code as needed.

Reply:
Took a quick look at this. This evaluated list is used multiple times in the section below. I would need to dive deeper into the `get_pvalue()` code to see exactly what is needed and what can be removed. Let's save this for a separate pull request whose aim is to improve efficiency 

```r
 survdifflist_eval <-
    lapply(survdifflist, eval, envir = attr(survfit_object$call, ".Environment"))

# Statlist ----------------------------------------------------------------

  statlist <- unique(statlist)
  statlist <- base::sub("test", "Equality across strata", statlist, fixed = TRUE)
  statlist <- base::sub("pvalue", "p-value", statlist, fixed = TRUE)
  Nms <- names(survdifflist_eval)

  stat_summary <- list(
    `Equality across strata` = rlang::expr(base::sub(
      "Custom",
      paste0("Harrington and Fleming test (rho = ", rho, ")"),
      Nms,
      fixed = TRUE
    )),
    `Chisq`    = rlang::expr(unlist(
      lapply(survdifflist_eval, function(x)
        format(round(x$chisq, 3), nsmall = 3, justify = "right", width = 6, scientific = FALSE)
      ))),
    df        = rlang::expr(unlist(
      lapply(survdifflist_eval, function(x)
        length(x$n) - 1)
    )),
    `p-value` = rlang::expr(unlist(
      lapply(survdifflist_eval, function(x)
        .pvalformat(
          stats::pchisq(x$chisq, length(x$n) - 1, lower.tail = FALSE)
        ))
    ))
  )[statlist]
```

Reply:
OK but then put the original code back. How it is written originally, it create an unevaluated list and only evaluated the required elements.

Reply:
The update I made was to evaluate in the correct environment. You'll need to explain to me how the previous code only evaluated a single call. Let's either join Mondays call early to discuss or stay on after.